tests/tests/test_filetransfer.py

```diff
@@ -51,7 +56,23 @@ def download_file(path, devid, authtoken):
     )
     download_url = "%s/devices/%s/download" % (deviceconnect_url, devid)
     download_url_with_path = download_url + "?path=" + urllib.parse.quote(path)
-    return requests.get(download_url_with_path, verify=False, headers=authtoken)
+    max_tries = 255
+    while max_tries > 0:
+        r = requests.get(download_url_with_path, verify=False, headers=authtoken)
+        if r.status_code == 503:
+            logger.info(
+                "test_filetransfer::download_file: got 503; retrying %d" % max_tries
+            )
+        else:
+            break
+        max_tries = max_tries - 1
+        time.sleep(1)
+    if r.status_code == 503:
+        logger.info(
+            "test_filetransfer::download_file: after all the tries, still: 503; returning"
+        )
+
+    return r
```

Comment:
@tranchitella another one I had to add. there is another unstability in the tests (not related to the current PR).